openpype/pipeline/publish/publish_plugins.py

```diff
@@ -380,64 +349,149 @@ def set_representation_colorspace(
-        # check extension
-        self.log.debug("__ ext: `{}`".format(ext))
+        # using cached settings if available
+        set_colorspace_data_to_representation(
+            representation, context.data,
+            colorspace,
+            colorspace_settings,
+            log=self.log
+        )
 
-        # check if ext in lower case is in self.allowed_ext
-        if ext.lstrip(".").lower() not in self.allowed_ext:
-            self.log.debug(
-                "Extension '{}' is not in allowed extensions.".format(ext)
-            )
-            return
 
-        if colorspace_settings is None:
-            colorspace_settings = self.get_colorspace_settings(context)
+class PrepRepresentationPluginMixin:
+    """Mixin for preparation of representation plugins.
+
+    This mixin provides methods for farm plugins to use.
+    """
 
-        # in case host color management is not enabled
-        if not colorspace_settings:
-            self.log.warning("Host's colorspace management is disabled.")
-            return
+    def set_expected_files_to_instance(
+        self,
+        instance,
+        frame_start,
+        frame_end,
+        path,
+        only_existing=False
+    ):
+        """Create expected files in instance data
 
-        # unpack colorspace settings
-        config_data, file_rules = colorspace_settings
+        Args:
+            instance (pyblish.api.Instance): Instance to set expected files on
+            frame_start (int): Start frame of the sequence
+            frame_end (int): End frame of the sequence
+            path (str): Path to generate expected files from
+            only_existing (Optional[bool]): Ensure that files exists.
 
-        if not config_data:
-            # warn in case no colorspace path was defined
-            self.log.warning("No colorspace management was defined")
-            return
+        Returns:
+            None: sets `expectedFiles` key on instance data
+        """
 
-        self.log.debug("Config data is: `{}`".format(config_data))
+        expected_files_list = file_collection.generate_expected_filepaths(
+            frame_start, frame_end, path, only_existing)
+        instance.data["expectedFiles"] = expected_files_list
 
-        project_name = context.data["projectName"]
-        host_name = context.data["hostName"]
-        project_settings = context.data["project_settings"]
+    def add_farm_instance_data(self, instance):
+        """Add farm publishing related instance data.
 
-        # get one filename
-        filename = representation["files"]
-        if isinstance(filename, list):
-            filename = filename[0]
+        Args:
+            instance (pyblish.api.Instance): pyblish instance
+        """
 
-        self.log.debug("__ filename: `{}`".format(filename))
+        # make sure rendered sequence on farm will
+        # be used for extract review
+        instance.data["useSequenceForReview"] = bool(
+            instance.data.get("review"))
```

Comment:
Review of instance is define by `"review"` family. As metioned we wanted to get rid of the key `"review"` for a very long time, this is adding new, same, key that we actually don't want.